### ISSUE TITLE
IPM 0.9 dependency resolution taking a long time

### ISSUE THREAD

**Issue by isc-shuliu** [intersystems/ipm#652]
Some packages will sporadically take a long time to install on the `v1` branch which has IPM 0.9.x.

## Steps to Reproduce
To reproduce this issue:

1. git clone the repo and switch to the `v1` branch
2. `docker compose -d --build`
3. `docker exec -it ipm-iris-1 bash`
4. Inside the container, run `iris session iris`
5. In the IRIS session, run `zpm "install -v demo-pandas-analytics"`
6. Observe the output, for me, it keeps hanging after the following output:

```
Module demo-pandas-analytics was downloaded from registry (https://pm.community.intersystems.com/)
Loading from /usr/irissys/mgr/.modules/USER/demo-pandas-analytics/1.0.0/

Load started on 12/10/2024 19:47:39
Loading file /usr/irissys/mgr/.modules/USER/demo-pandas-analytics/1.0.0/module.xml as xml
Imported document: demo-pandas-analytics.ZPM
Load finished successfully.

Dependencies:
dsw 3.2.26 @ registry
mdx2json 3.2.41 @ registry
yaml-utils 0.1.4 @ registry
zpm-registry 1.3.2 @ registry
```

7. Press `Ctrl`+`p` followed by `Ctrl` + `q` to detach from the container. 
8. Run `docker compose down && docker compose up -d --build &&  docker exec -it ipm-iris-1 /bin/bash` to start a fresh container
9. Run `iris session iris` to start an IRIS session
10. In the IRIS session, run the following commands to install dependencies first
  * `zpm "install -v dsw 3.2.26"`
  * `zpm "install -v mdx2json 3.2.41"`
  * `zpm "install -v yaml-utils 0.1.4"`
  * `zpm "install -v zpm-registry 1.3.2"`
 11. Now `zpm "install -v demo-pandas-analytics"` finishes within a reasonable period of time.

**Comment by isc-shuliu** [comment]
A number of packages are influenced by this bug, including
* demo-pandas-analytics
* iris-flow
* telegram-adapter-demo
* openapi-suite
* iris-flow

**Comment by isc-tleavitt** [comment]
This is a deadlock in %SYS.NAMESPACE resulting from https://github.com/intersystems/ipm/commit/6ed90b1622ecb424240db583b102050e289ba26c - the logic here is very flawed, we explicitly add mappings for *everything* because of course the item doesn't exist yet, we're loading it for the first time.

This needs to be fixed for 0.9 release and may impact HS adoption.

cc @isc-kiyer @isc-eneil @isc-jlechtne @isc-jili

**Comment by isc-kiyer** [comment]
@isc-tleavitt thanks for flagging that. We would not run into it for hscore since we use -DNoMapping=1 but would for other components. Also, the code should probably use `%SYS.Namespace:Get*Dest` methods to check if the item would be loaded into the expected database instead of already checking existence.
